api/src/main/scala/hmda/api/http/admin/PublicationAdminHttpApi.scala

```diff
@@ -0,0 +1,51 @@
+package hmda.api.http.admin
+
+import akka.actor.{ ActorRef, ActorSystem }
+import akka.event.LoggingAdapter
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives._
+import akka.pattern.ask
+import akka.stream.ActorMaterializer
+import akka.util.Timeout
+import hmda.api.http.HmdaCustomDirectives
+import hmda.api.protocol.processing.ApiErrorProtocol
+import hmda.model.fi.SubmissionId
+import hmda.persistence.messages.commands.disclosure.DisclosureCommands.GenerateDisclosureReports
+
+import scala.util.{ Failure, Success }
+
+trait PublicationAdminHttpApi extends HmdaCustomDirectives with ApiErrorProtocol {
+
+  implicit val system: ActorSystem
+  implicit val materializer: ActorMaterializer
+  implicit val timeout: Timeout
+
+  val log: LoggingAdapter
+
+  def disclosureGenerationPath(supervisor: ActorRef) =
+    path("disclosure" / Segment / IntNumber / IntNumber) { (instId, year, subId) =>
```

Comment:
If the submission hasn't been signed, the data won't be in the LAR repository yet, so the reports will all be empty. Let's make sure that doesn't happen by returning a `400` if the submission isn't signed.

Reply:
Done in most recent commit